doc/flux_to_redux.md

```diff
@@ -0,0 +1,579 @@
+
+```dart
+// A regular Dart class that represents the application data model
+class ReduxState {
+    // All of the state fields for the application
+    String mainBackgroundColor;
+    String blockOneBackgroundColor;
+    String blockTwoBackgroundColor;
+    String blockThreeBackgroundColor;
+
+    // A constructor for creating the default app state
+    ReduxState.defaultState()
+        : this.mainBackgroundColor = 'gray',
+        this.blockOneBackgroundColor = 'red',
+        this.blockTwoBackgroundColor = 'orange',
+        this.blockThreeBackgroundColor = 'blue';
+
+    // A convenience constructor for creating new instances of the state class
+    ReduxState.update(ReduxState oldState,
+        {mainBackgroundColor, blockOneBackgroundColor, blockTwoBackgroundColor,         blockThreeBackgroundColor})
+        : this.mainBackgroundColor = mainBackgroundColor ?? oldState.mainBackgroundColor,
+        this.blockOneBackgroundColor = blockOneBackgroundColor ?? oldState.blockOneBackgroundColor,
+        this.blockTwoBackgroundColor = blockTwoBackgroundColor ?? oldState.blockTwoBackgroundColor,
+        this.blockThreeBackgroundColor = blockThreeBackgroundColor ?? oldState.blockThreeBackgroundColor;
+}
+```
+
+```dart
+// An action for updating each state property
+class UpdateBackgroundColorAction {}
+
+class UpdateBlockOneBackgroundColorAction {}
+
+class UpdateBlockTwoBackgroundColorAction {}
+
+class UpdateBlockThreeBackgroundColorAction {}
+```
+
+```dart
+// A simple reducer that has a case for every action.
+// Because the example is relatively contrived to keep things simple, there are other
+// ways this could be done given that all the actions are pretty much the same,
+// but the reducer was written more verbosely to capture the core idea.
+ReduxState afterTransitionReducer(ReduxState oldState, dynamic action) {
+  if (action is UpdateBackgroundColorAction) {
+    var color = '#' + (Random().nextDouble() * 16777215).floor().toRadixString(16);
+
+    return ReduxState.update(oldState, mainBackgroundColor: color);
+  } else if (action is UpdateBlockOneBackgroundColorAction) {
+    var color = '#' + (Random().nextDouble() * 16777215).floor().toRadixString(16);
+
+    return ReduxState.update(oldState, blockOneBackgroundColor: color);
+  } else if (action is UpdateBlockTwoBackgroundColorAction) {
+    var color = '#' + (Random().nextDouble() * 16777215).floor().toRadixString(16);
+
+    return ReduxState.update(oldState, blockTwoBackgroundColor: color);
+  } else if (action is UpdateBlockThreeBackgroundColorAction) {
+    var color = '#' + (Random().nextDouble() * 16777215).floor().toRadixString(16);
+
+    return ReduxState.update(oldState, blockThreeBackgroundColor: color);
+  }
+
+  return oldState;
+}
+```
+
+```dart
+// Instantiate a store using the reducer and state class
```

Comment:
I think we should show the necessary imports for clarity as to what the `redux` namespace represents.

```suggestion
import 'package:redux/redux.dart' as redux;
    
// Instantiate a store using the reducer and state class
```